internal/charmstore/resources.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmstore // import "gopkg.in/juju/charmstore.v5-unstable/internal/charmstore"
+
+import (
+	"io"
+	"time"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
+)
+
+var resourceNotFound = errgo.Newf("resource not found")
+
+// ListResources returns the list of resources for the charm at the
+// latest revision for each resource.
+func (s Store) ListResources(entity *mongodoc.Entity, channel params.Channel) ([]*mongodoc.Resource, error) {
```

Comment:
Could we call this ListLatestResources?  To differentiate from the inevitable ListResources that takes a specific charm-revision and channel and returns just what the charm revision tuple indicates?  I expected this to mirror the ListResources on the clientside, and it's quite different.